router_test.go

```diff
@@ -605,3 +605,53 @@ func TestRequestContentTypes(t *testing.T) {
 	assert.Equal(t, http.StatusOK, w.Code, w.Body.String())
 	assert.JSONEq(t, `{"name": "one two"}`, w.Body.String())
 }
+
+func TestOpenAPIOrdering(t *testing.T) {
+	app := newTestRouter()
+
+	type Response struct {
+		Description string `json:"description"`
+		Category    string `json:"category"`
+		ID          string `json:"id"`
+		Name        string `json:"name"`
+	}
+
+	app.Resource("/menu/{menu-category}/item/{item-id}").Get("cafe menu", "ISP Cafe",
+		NewResponse(http.StatusOK, "test").Model(Response{}),
+	).Run(func(ctx Context, input struct {
+		Category string `path:"menu-category"`
+		ItemID   string `path:"item-id"`
+	}) {
+		ctx.WriteModel(http.StatusOK, Response{
+			Category:    input.Category,
+			ID:          input.ItemID,
+			Name:        "Apple Tea",
+			Description: "Green tea with apples",
+		})
+	})
+
+	w := httptest.NewRecorder()
+	req, _ := http.NewRequest(http.MethodGet, "/menu/drinks/item/apple-tea", nil)
+	req.Header.Set("Authorization", "dummy")
+	req.Host = "example.com"
+	app.ServeHTTP(w, req)
+
+	// JSON response should be lexicographically sorted
+	assert.Equal(t, http.StatusOK, w.Code)
+	expectedResp := `{"$schema":"https://example.com/schemas/Response.json","category":"drinks","description":"Green tea with apples","id":"apple-tea","name":"Apple Tea"}`
+	assert.Equal(t, expectedResp, w.Body.String())
+
+	// Parameters should match insertion order
+	openapi := app.OpenAPI().Search("paths", "/menu/{menu-category}/item/{item-id}", "get").Bytes()
+	type parameters struct {
+		Name string `json:"name"`
+	}
+
+	type opschema struct {
+		Parameters []parameters `json:"parameters"`
+	}
+
+	var p opschema
+	json.Unmarshal(openapi, &p)
+	assert.Equal(t, p.Parameters, []parameters{{Name: "menu-category"}, {Name: "item-id"}})
```

Comment:
non-blocking: there's a ['JSONEq'](https://pkg.go.dev/github.com/stretchr/testify/assert#JSONEq) that could be useful here. Something like:

```go
assert.JSONEq(t, `[{"name": "menu-category", ...}]`, openapi)`
```

Reply:
I considered using that, but unfortunately, the output of the `gabs.Search` ends at the operation level.  This results in `openapi` containing the full spec for the `get` operation (ex. `responses`, additional properties for a `parameter`). `JSONEq` only returns true if the entire JSON payload matches (ignoring order).  Given we're only interested in using the `name` to check order was correctly preserved, I decided with a temp type to only parse the necessary field.